src/profile_manager/views.py

```diff
@@ -115,6 +116,9 @@ def get_context_data(self, **kwargs):
         context['xp_per_course'] = profile.xp_per_course()
         context['badge_assertions_dict_items'] = BadgeAssertion.objects.badge_assertions_dict_items(profile.user)
 
+        # get tags and its xp that are related to user + sort by xp total
+        context['tags'] = sorted([(tag, total_xp_by_tags(profile.user, [tag])) for tag in get_tags_from_user(profile.user)], key=lambda el: el[1])
```

Comment:
Can you move this to a method on the Tags app, and preferably, expand it out so it is commented, readable, and tested? So much code crammed into a single line is unreadable.

I would much prefer loops and variables than everything crammed into a single line.  Space is free, there is usually little gained by trying to cram more into less space, and it usually violated the implicit vs explicit and simple vs complex.

Reply:
done 
![image](https://user-images.githubusercontent.com/39788517/182747709-b326eaf2-e82e-410d-96c0-7b713539941b.png)

Reply:
Awesome! 

From your code comments and docstring it's not clear HOW the data is sorted. (By XP? By alphabetical tag names?)

Reply:
added ``Sorted by total xp in descending order``

added ``(descending order)`` to ``# sort by total xp then return``